src/Rings/groebner.jl

```diff
@@ -419,29 +422,297 @@ Singular Polynomial Ring (QQ),(a,b,c),(dp(3),C)
 julia> I = Singular.Ideal(SR,[SR(-1+c+b+a^3),SR(-1+b+c*a+2*a^3),SR(5+c*b+c^2*a)])
 Singular ideal over Singular Polynomial Ring (QQ),(a,b,c),(dp(3),C) with generators (a^3 + b + c - 1, 2*a^3 + a*c + b - 1, a*c^2 + b*c + 5)
 
-julia> Oscar.normal_form_internal(I,J)
+julia> Oscar.normal_form_internal(I,J,default_ordering(base_ring(J)))
 3-element Vector{fmpq_mpoly}:
  a^3
  2*a^3 + 2*a - 2*c
  4*a - 2*c^2 - c + 5
 ```
 """
-function normal_form_internal(I::Singular.sideal, J::MPolyIdeal)
-  groebner_assure(J)
-  G = collect(values(J.gb))[1]
-  singular_assure(G)
+function normal_form_internal(I::Singular.sideal, J::MPolyIdeal, o::MonomialOrdering)
+  groebner_assure(J, o)
+  G = J.gb[o]  
+  singular_assure(G, o)
   K = ideal(base_ring(J), reduce(I, G.S))
   return [J.gens.Ox(x) for x = gens(K.gens.S)]
 end
 
 @doc Markdown.doc"""
-    normal_form(f::T, J::MPolyIdeal) where { T <: MPolyElem }
+	reduce(I::IdealGens, J::IdealGens; ordering::MonomialOrdering = default_ordering(base_ring(J)))
+
+Return a `Vector` whose elements are the underlying elements of `I`
+reduced by the underlying generators of `J` w.r.t. the monomial
+ordering `ordering`. `J` need not be a Groebner basis. The returned
+`Vector` will have the same number of elements as `I`, even if they
+are zero.
+
+	reduce(f::T, F::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(f))) where {T <: MPolyElem}
+
+Return an element which is `f` reduced by the underlying generators
+of `F` w.r.t. the monomial ordering `ordering`. `F` need not be a
+Groebner basis. The returned
```

Comment:
There seems to be half a sentence missing (or too many). Maybe a Cut&Paste error.

Reply:
should be fixed in https://github.com/oscar-system/Oscar.jl/pull/1654/commits/1e9bf431fb77d5d593b32b587c0093ad48cbf928